packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -507,12 +507,27 @@ class TextStyle with Diagnosticable {
        assert(backgroundColor == null || background == null, _kColorBackgroundWarning);
 
 
-  /// Whether null values are replaced with their value in an ancestor text
-  /// style (e.g., in a [TextSpan] tree).
+  /// Whether null values in this [TextStyle] can be replaced with their value
+  /// in another [TextStyle] using [merge].
   ///
-  /// If this is false, properties that don't have explicit values will revert
-  /// to the defaults: white in color, a font size of 14 pixels, in a sans-serif
-  /// font face.
+  /// The [merge] operation is not commutative: the [inherit] value of the
+  /// method argument decides whether the two [TextStyle]s can be combined
+  /// together. If it is false, the method argument [TextStyle] will be returned.
+  /// Otherwise, the combining is allowed, and the returned [TextStyle] inherits
+  /// the [inherit] value from the method receiver.
+  ///
+  /// This property has no effect on [TextSpan]'s text style cascading: in a
+  /// [TextSpan] tree, a [TextSpan]'s text style can be combined with that of an
+  /// ancestor [TextSpan] if it has unspecified fields, regardless of its
+  /// [inherit] value.
+  ///
+  /// Properties that don't have explicit values or other default values to fall
+  /// back to will revert to the defaults defined in the text shaper: white in
```

Comment:
"text shaper"? Will developers know what that refers to? Unless "text shaper" refers to something that will make this point easier to understand, just explaining what the defaults are is probably enough.

Note also: I'm surprised that the default color is white (instead of black).

Reply:
I mostly copied these defaults without actually checking (the font size is correct). It could be black let me check.

Reply:
yeah it's correct. Should I say "defined in ui.Paragraph" instead?

Reply:
got rid of "text shaper"